docs/experimental-features.md

```diff
@@ -218,36 +218,86 @@ configured, the daemon will fail to start.
 ---
 
 ## ipfs p2p
-Allows to tunnel TCP connections through Libp2p streams
+
+Allows tunneling of TCP connections through Libp2p streams. If you've ever used
+port forwarding with SSH (the `-L` option in openssh), this feature is quite
+similar.
 
 ### State
+
 Experimental
 
 ### In Version
+
 master, 0.4.10
 
 ### How to enable
-P2P command needs to be enabled in config
 
-`ipfs config --json Experimental.Libp2pStreamMounting true`
+The `p2p` command needs to be enabled in config:
+
+```sh
+> ipfs config --json Experimental.Libp2pStreamMounting true
+```
 
 ### How to use
 
-Basic usage:
+First, pick a protocol name for your application. Think of the protocol name as
+a port number, just significantly more user-friendly. In this example, we're
+going to use `/kickass/1.0`.
+
+**Setup:**
+
+1. A "server" node with peer ID `$SERVER_ID`
+2. A "client" node.
+
+**On the "server" node:**
+
+First, start your application and have it listen on `$APP_PORT`.
+
+Then, configure the p2p listener by running:
+
+```sh
+> ipfs p2p forward /kickass/1.0 /ipfs /ip4/127.0.0.1/tcp/$APP_PORT
+```
+
+This will configure IPFS to forward all incoming `/p2p/kickass/1.0` streams to
+`127.0.0.1:$APP_PORT` (opening a new connection to `127.0.0.1:$APP_PORT` per
+incoming stream.
+
+**On the "client" node:**
+
+First, configure the p2p dialer to forward all inbound connections on
+`127.0.0.1:SOME_PORT` to the listener behind `/p2p/kickass/1.0` on the server
+node.
```

Comment:
>First, configure the p2p dialer to forward all inbound connections on `127.0.0.1:SOME_PORT`  to the listener behind `/p2p/kickass/1.0` on the server node.

It may be fine as is, but I wonder if this is more clear.

>First, configure the client p2p dialer, so that it forwards all inbound connections on `127.0.0.1:SOME_PORT` to the server node listening on `/p2p/kickass/1.0`.